packages/react-dom/src/events/ReactDOMEventListener.js

```diff
@@ -141,7 +149,9 @@ function handleTopLevel(bookKeeping: BookKeepingInstance) {
     if (!root) {
       break;
     }
-    bookKeeping.ancestors.push(ancestor);
+    if (ancestor.tag === HostComponent || ancestor.tag === HostText) {
```

Comment:
Maybe create a binding for tag here rather than doing the property lookup twice?